src/frontend/src/optimizer/plan_node/stream_hash_join.rs

```diff
@@ -45,6 +45,10 @@ pub struct StreamHashJoin {
     /// non-equal parts to facilitate execution later
     eq_join_predicate: EqJoinPredicate,
 
+    /// `(generate_output, key_required_larger, key_required_smaller, delta_expression)`
+    /// View struct `InequalityInputPair` for details.
+    inequality_pairs: Vec<(bool, InequalityInputPair)>,
```

Comment:
> IMO, we can maintain two valid `InequalityInputPair` instead of a `Vec<(bool, InequalityInputPair)>` here, because more than two `InequalityInputPair` seems unusual.

just fixed so that we now have at most 2 inequality conditions which are doing state cleaning, and won't permit inequality conditions to do state cleaning when equality condition can do.